src/test/java/com/wcc/platform/factories/TestFactories.java

```diff
@@ -10,140 +10,181 @@
+    try {
+      String content = FileUtil.readFileAsString(fileName);
+      return ObjectMapperTestFactory.getInstance().readValue(content, CollaboratorPage.class);
+    } catch (JsonProcessingException e) {
+      return createCollaboratorPageTest();
+    }
+  }
+
+  public static CodeOfConductPage createCodeOfConductPageTest() {
+    return new CodeOfConductPage(createPageTest(), List.of(createSectionTest()));
+  }
+
+  public static CodeOfConductPage createCodeOfConductPageTest(String fileName) {
+    try {
+      String content = FileUtil.readFileAsString(fileName);
+      return ObjectMapperTestFactory.getInstance().readValue(content, CodeOfConductPage.class);
+    } catch (JsonProcessingException e) {
+      return createCodeOfConductPageTest();
+    }
+  }
+
+  public static MemberByType createMemberByTypeTest() {
+    var directors = List.of(createLeadershipMemberTest(MemberType.DIRECTOR));
+    var leaders = List.of(createLeadershipMemberTest(MemberType.LEADER));
+    var evangelist = List.of(createLeadershipMemberTest(MemberType.EVANGELIST));
+    return new MemberByType(directors, leaders, evangelist);
+  }
+
+  public static Member createCollaboratorsTest() {
+    return (createCollaboratorMemberTest(MemberType.MEMBER));
+  }
+
+  public static Page createPageTest() {
+    return new Page("title", "subtitle", "description");
+  }
+
+  public static Section createSectionTest() {
+    return new Section("title", "description", List.of("item_1", "item_2", "item_3"));
+  }
+
+  public static Member createMemberTest(MemberType type) {
+    var member =
+        Member.builder()
+            .fullName("fullName " + type.name())
+            .position("position " + type.name())
+            .email("member@wcc.com")
+            .country(new Country("Country code", "Country name"))
+            .city("City")
+            .jobTitle("Job title")
+            .companyName("Company name")
+            .memberType(type)
+            .images(List.of(new Image("image.png", "alt image", ImageType.DESKTOP)))
+            .network(List.of(new SocialNetwork(SocialNetworkType.LINKEDIN, "collaborator_link")))
+            .build();
+
+    return member;
+  }
+
+  public static LeadershipMember createLeadershipMemberTest(MemberType type) {
+    //    var team = new LeadershipMember();
+    // Member member = createMemberTest(type);
+    LeadershipMember team =
+        LeadershipMember.leadershipMemberBuilder()
+            .fullName("fullName " + type.name())
+            .position("position " + type.name())
+            .email("member@wcc.com")
+            .country(new Country("Country code", "Country name"))
+            .city("City")
+            .jobTitle("Job title")
+            .companyName("Company name")
+            .memberType(type)
+            .images(List.of(new Image("image.png", "alt image", ImageType.DESKTOP)))
+            .network(List.of(new SocialNetwork(SocialNetworkType.LINKEDIN, "collaborator_link")))
+            .build();
```

Comment:
Above problematic code -  it is not possible to cast from Member to LeadershipMember in a simple way - it needs some playing around with the lombok annotations which are a bit confusing to me.  I have also hoped that I could reuse the initialization above (Member object) for Collaborators and leadershipMembers ...